### ISSUE TITLE
Multiple TypeScript TS2611 errors when building with 3.0.1

### ISSUE THREAD

**Issue by kevinmkane** [PeculiarVentures/ASN1.js#75]
I'm trying to use asn1js in a new TypeScript project of mine, and when I run the build I get the following errors. I'm using the published 3.0.1 npm package. I can still run my test program afterwards and successfully decode an ASN.1 object, though. Are these errors expected?

```
node_modules/asn1js/build/index.d.ts:273:9 - error TS2611: 'valueBeforeDecode' is defined as a property in class '{ isHexOnly: boolean; valueHex: ArrayBuffer; valueHexView: Uint8Array; fromBER(inputBuffer: ArrayBuffer | Uint8Array, inputOffset: number, inputLength: number): number; ... 6 more ...; valueBeforeDecodeView: Uint8Array; } & LocalBaseBlock', but is overridden here in 'LocalIdentificationBlock' as an accessor.

273     get valueBeforeDecode(): ArrayBuffer;
            ~~~~~~~~~~~~~~~~~

node_modules/asn1js/build/index.d.ts:283:9 - error TS2611: 'valueHex' is defined as a property in class '{ isHexOnly: boolean; valueHex: ArrayBuffer; valueHexView: Uint8Array; fromBER(inputBuffer: ArrayBuffer | Uint8Array, inputOffset: number, inputLength: number): number; ... 6 more ...; valueBeforeDecodeView: Uint8Array; } & LocalBaseBlock', but is overridden here in 'LocalIdentificationBlock' as an accessor.

283     set valueHex(v: ArrayBuffer);
            ~~~~~~~~

node_modules/asn1js/build/index.d.ts:420:9 - error TS2611: 'valueBeforeDecode' is defined as a property in class '{ isHexOnly: boolean; valueHex: ArrayBuffer; valueHexView: Uint8Array; fromBER(inputBuffer: ArrayBuffer | Uint8Array, inputOffset: number, inputLength: number): number; ... 6 more ...; valueBeforeDecodeView: Uint8Array; } & ValueBlock', but is overridden here in 'LocalStringValueBlock' as an accessor.

420     get valueBeforeDecode(): ArrayBuffer;
            ~~~~~~~~~~~~~~~~~

node_modules/asn1js/build/index.d.ts:430:9 - error TS2611: 'valueHex' is defined as a property in class '{ isHexOnly: boolean; valueHex: ArrayBuffer; valueHexView: Uint8Array; fromBER(inputBuffer: ArrayBuffer | Uint8Array, inputOffset: number, inputLength: number): number; ... 6 more ...; valueBeforeDecodeView: Uint8Array; } & ValueBlock', but is overridden here in 'LocalStringValueBlock' as an accessor.

430     set valueHex(v: ArrayBuffer);
            ~~~~~~~~

node_modules/asn1js/build/index.d.ts:503:9 - error TS2611: 'valueBeforeDecode' is defined as a property in class '{ isHexOnly: boolean; valueHex: ArrayBuffer; valueHexView: Uint8Array; fromBER(inputBuffer: ArrayBuffer | Uint8Array, inputOffset: number, inputLength: number): number; ... 6 more ...; valueBeforeDecodeView: Uint8Array; } & ValueBlock', but is overridden here in 'LocalPrimitiveValueBlock' as an accessor.

503     get valueBeforeDecode(): ArrayBuffer;
            ~~~~~~~~~~~~~~~~~

node_modules/asn1js/build/index.d.ts:513:9 - error TS2611: 'valueHex' is defined as a property in class '{ isHexOnly: boolean; valueHex: ArrayBuffer; valueHexView: Uint8Array; fromBER(inputBuffer: ArrayBuffer | Uint8Array, inputOffset: number, inputLength: number): number; ... 6 more ...; valueBeforeDecodeView: Uint8Array; } & ValueBlock', but is overridden here in 'LocalPrimitiveValueBlock' as an accessor.

513     set valueHex(v: ArrayBuffer);
            ~~~~~~~~

node_modules/asn1js/build/index.d.ts:642:9 - error TS2611: 'valueBeforeDecode' is defined as a property in class '{ isHexOnly: boolean; valueHex: ArrayBuffer; valueHexView: Uint8Array; fromBER(inputBuffer: ArrayBuffer | Uint8Array, inputOffset: number, inputLength: number): number; ... 6 more ...; valueBeforeDecodeView: Uint8Array; } & ValueBlock', but is overridden here in 'LocalBooleanValueBlock' as an accessor.

642     get valueBeforeDecode(): ArrayBuffer;
            ~~~~~~~~~~~~~~~~~

node_modules/asn1js/build/index.d.ts:652:9 - error TS2611: 'valueHex' is defined as a property in class '{ isHexOnly: boolean; valueHex: ArrayBuffer; valueHexView: Uint8Array; fromBER(inputBuffer: ArrayBuffer | Uint8Array, inputOffset: number, inputLength: number): number; ... 6 more ...; valueBeforeDecodeView: Uint8Array; } & ValueBlock', but is overridden here in 'LocalBooleanValueBlock' as an accessor.

652     set valueHex(v: ArrayBuffer);
            ~~~~~~~~

node_modules/asn1js/build/index.d.ts:726:9 - error TS2611: 'valueBeforeDecode' is defined as a property in class '{ isHexOnly: boolean; valueHex: ArrayBuffer; valueHexView: Uint8Array; fromBER(inputBuffer: ArrayBuffer | Uint8Array, inputOffset: number, inputLength: number): number; ... 6 more ...; valueBeforeDecodeView: Uint8Array; } & LocalConstructedValueBlock', but is overridden here in 'LocalOctetStringValueBlock' as an accessor.

726     get valueBeforeDecode(): ArrayBuffer;
            ~~~~~~~~~~~~~~~~~

node_modules/asn1js/build/index.d.ts:736:9 - error TS2611: 'valueHex' is defined as a property in class '{ isHexOnly: boolean; valueHex: ArrayBuffer; valueHexView: Uint8Array; fromBER(inputBuffer: ArrayBuffer | Uint8Array, inputOffset: number, inputLength: number): number; ... 6 more ...; valueBeforeDecodeView: Uint8Array; } & LocalConstructedValueBlock', but is overridden here in 'LocalOctetStringValueBlock' as an accessor.

736     set valueHex(v: ArrayBuffer);
            ~~~~~~~~

node_modules/asn1js/build/index.d.ts:808:9 - error TS2611: 'valueBeforeDecode' is defined as a property in class '{ isHexOnly: boolean; valueHex: ArrayBuffer; valueHexView: Uint8Array; fromBER(inputBuffer: ArrayBuffer | Uint8Array, inputOffset: number, inputLength: number): number; ... 6 more ...; valueBeforeDecodeView: Uint8Array; } & LocalConstructedValueBlock', but is overridden here in 'LocalBitStringValueBlock' as an accessor.

808     get valueBeforeDecode(): ArrayBuffer;
            ~~~~~~~~~~~~~~~~~

node_modules/asn1js/build/index.d.ts:818:9 - error TS2611: 'valueHex' is defined as a property in class '{ isHexOnly: boolean; valueHex: ArrayBuffer; valueHexView: Uint8Array; fromBER(inputBuffer: ArrayBuffer | Uint8Array, inputOffset: number, inputLength: number): number; ... 6 more ...; valueBeforeDecodeView: Uint8Array; } & LocalConstructedValueBlock', but is overridden here in 'LocalBitStringValueBlock' as an accessor.

818     set valueHex(v: ArrayBuffer);
            ~~~~~~~~

node_modules/asn1js/build/index.d.ts:888:9 - error TS2611: 'valueBeforeDecode' is defined as a property in class '{ isHexOnly: boolean; valueHex: ArrayBuffer; valueHexView: Uint8Array; fromBER(inputBuffer: ArrayBuffer | Uint8Array, inputOffset: number, inputLength: number): number; ... 6 more ...; valueBeforeDecodeView: Uint8Array; } & ValueBlock', but is overridden here in 'LocalIntegerValueBlock' as an accessor.

888     get valueBeforeDecode(): ArrayBuffer;
            ~~~~~~~~~~~~~~~~~

node_modules/asn1js/build/index.d.ts:898:9 - error TS2611: 'valueHex' is defined as a property in class '{ isHexOnly: boolean; valueHex: ArrayBuffer; valueHexView: Uint8Array; fromBER(inputBuffer: ArrayBuffer | Uint8Array, inputOffset: number, inputLength: number): number; ... 6 more ...; valueBeforeDecodeView: Uint8Array; } & ValueBlock', but is overridden here in 'LocalIntegerValueBlock' as an accessor.

898     set valueHex(v: ArrayBuffer);
            ~~~~~~~~

node_modules/asn1js/build/index.d.ts:991:9 - error TS2611: 'valueBeforeDecode' is defined as a property in class '{ isHexOnly: boolean; valueHex: ArrayBuffer; valueHexView: Uint8Array; fromBER(inputBuffer: ArrayBuffer | Uint8Array, inputOffset: number, inputLength: number): number; ... 6 more ...; valueBeforeDecodeView: Uint8Array; } & ValueBlock', but is overridden here in 'LocalSidValueBlock' as an accessor.

991     get valueBeforeDecode(): ArrayBuffer;
            ~~~~~~~~~~~~~~~~~

node_modules/asn1js/build/index.d.ts:1001:9 - error TS2611: 'valueHex' is defined as a property in class '{ isHexOnly: boolean; valueHex: ArrayBuffer; valueHexView: Uint8Array; fromBER(inputBuffer: ArrayBuffer | Uint8Array, inputOffset: number, inputLength: 
number): number; ... 6 more ...; valueBeforeDecodeView: Uint8Array; } & ValueBlock', but is overridden here in 'LocalSidValueBlock' as an accessor.
```

**Comment by microshine** [comment]
Could you share your tsconfig.json file? I can't reproduce this error. I use the latest version in our projects without any TS error

**Comment by kevinmkane** [comment]
Absolutely.

```
{
  "include": [
    "validatortest.ts",
    "types.d.ts",
    "pkijs.d.ts",
    "lib/*.ts",
    "lib/**/*.ts"
  ],
  "exclude": ["node_modules"],
  "compilerOptions": {
    /* Basic Options */
    "target": "es2015",                                /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019', 'ES2020', 'ES2021', or 'ESNEXT'. */
    "module": "es2015",                           /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', 'es2020', or 'ESNext'. */
    "outDir": "dist",                             /* Concatenate and emit output to single file. */
    /* Strict Type-Checking Options */
    "strict": true,                                 /* Enable all strict type-checking options. */
    "esModuleInterop": true,                        /* Enables emit interoperability between CommonJS and ES Modules via creation of namespace objects for all imports. Implies 'allowSyntheticDefaultImports'. */
    "forceConsistentCasingInFileNames": true,       /* Disallow inconsistently-cased references to the same file. */
    "sourceMap": true,
    "moduleResolution": "node",
    "lib" : [
      "es2015",
      "dom"
    ]
  }
}
```

**Comment by microshine** [comment]
Thanks. I'll test in an hour

**Comment by kevinmkane** [comment]
Oh, I should mention I'm _attempting_ to do this on Windows, so there is the possibility that's introducing some strangeness.

**Comment by kevinmkane** [comment]
I do still see this if I build my project inside a WSL Ubuntu session.

**Comment by microshine** [comment]
I've published `asn1js@3.0.2`. Please try it

**Comment by kevinmkane** [comment]
The errors are gone now. Thanks!
